src/Units/UnitState.java

```diff
@@ -80,6 +80,11 @@ public void fire(WeaponModel weapon)
     }
   }
 
+  public boolean hasMaterials()
+  {
+    return materials > 0 || materials == model.maxMaterials;
```

Comment:
Why `materials == model.maxMaterials`? Won't that result in `hasMaterials()` returning true for units that don't have any?

Reply:
That's... true, actually. What are you doing giving material-using actions to unit types with 0 `maxMaterials`, though?

Reply:
And you need the `==` comparison for the -1/inf case, of course. I think I've grown towards using dedicated booleans instead of sentinel values just to avoid obtuse logic like this.

Reply:
You say boolean - can I raise you a `moveFuelBurn` integer (matching `idleFuelBurn`) that can be 0-1 (or go up to 2 to support DS Olaf or shenanigans units)?

Reply:
> FuelBurn

Sounds like a separate discussion, but: Do we have units that burn fuel differently than the movement values? Depending on the use though, sure.

Reply:
Fuel is burned 3 ways:
Drake powers
Idle burn (sea/air)
1 fuel per move point spent

Snow in Dual Strike increases fuel burn to 2 per move point spent.